CHANGELOG.md

```diff
@@ -17,6 +17,7 @@
 * Core should not use the version passed in `Realm` constructor to load the schema at some particular version, but only for setting `m_frozen_version`.
 * Fixed possible segfault in sync client where async callback was using object after being deallocated ([#6053](https://github.com/realm/realm-core/issues/6053), since v11.7.0)
 * Fixed crash when using client reset with recovery and flexible sync with a single subscription ([#6070](https://github.com/realm/realm-core/issues/6070), since v12.3.0)
+* Fixed crash with wrong transaction state, when using client reset with discard local mode ([#6144](https://github.com/realm/realm-core/issues/6144), since v13.0.0)
```

Comment:
It might be helpful to tell SDKs that this is a general problem for `Realm::freeze()`, and only relates to client reset because that happens to use frozen Realms.